### ISSUE TITLE
bug(fxLayout, universal): layout direction is wrong

### ISSUE THREAD

**Issue by CaerusKaru** [angular/flex-layout#423]
When using Universal rendering, fxLayout gets picked up, but wrongly applied as `flex-direction=row` instead of `flex-direction=column`.

Confirmed that this only happens on the latest commit, which means a wire got crossed somewhere in the refactoring for NgClass and NgStyle (e461d17471f713deaf4663e0a0438822f8f0ad86).

**Comment by ThomasBurleson** [comment]
There are known issues with current @angular/flex-layout + Universal

*  https://github.com/angular/flex-layout/issues/373

>  Please provide an HTML code sample to verify your issue is different from #373.

**Comment by CaerusKaru** [comment]
@ThomasBurleson There are two reasons I didn't file this under #373:

1. The issues only cropped up for me after #419 was merged
2. That issue looked like it pertained exclusively to breakpoint management in Universal, whereas this issue is just a misapplied CSS styling for all cases

The HTML template looks like this:
```
<div fxFill fxLayout="column">
  <div class="nav-bar">
    <nav md-tab-nav-bar fxFlex>
        ... nav bar elements
```
and the top-level div gets rendered with the following CSS with Universal:
```
width: 100%;
height: 100%;
min-width: 100%;
min-height: 100%;
display: flex;
box-sizing: border-box;
flex-direction: row;
-webkit-flex-direction: row;
```
There are no other `fxFlex`-related elements anywhere else in this template, though there is a `<router-outlet>` that inevitably contains other flex elements.

**Comment by ThomasBurleson** [comment]
@CaerusKaru - #419 fixed an performance/rendering issue where each/every `class=""` attribute created ClassDirective instance. Now only `[ngClass]` and `[ngStyle]` attributes will trigger ClassDirective or StyleDirective instantiations (respectively).

#419 is an issue with Universal when using the Responsive API. Your usage is with the Static API... which should render the correct CSS. 

> We will investigate.

**Comment by alirezamirian** [comment]
@CaerusKaru 
Not sure this helps, but I had [a similar issue](https://github.com/angular/flex-layout/issues/389),
but when I tried to reproduce it, it turned out that the cause of the problem was an error somewhere else, not related to flex-layout! I don't know why `flex-direction: row` where applying instead of not working at all, and I didn't further investigate it.

**Comment by CaerusKaru** [comment]
@alirezamirian While I'd like to believe that it's an error on my end, and thus easily correctable, that doesn't explain why the rendering works with versions of flex-layout prior to PR #419

**Comment by CaerusKaru** [comment]
Here's the post-mortem on this issue. The issue only comes up when there is a `fxFlex` descendant of a `fxLayout`. This is because the `fxFlex` tries to mutate its parent if it doesn't already have a `flex-direction` set, and bases its own properties on this if it does exist. 

By setting the default for the `BehaviorSubject` in `fxLayout` to be `row`, it wasn't getting updated because Universal had finished rendering at that point, and didn't need to revisit the subscription. By changing this to a `ReplaySubject`, we get the benefit of storing the most recent value, but we also prevent the subscription from returning a fake value. Instead, the children will get the correct value every time. 

This might be the same issue that cropped up in #373, since their template involves a `fxFlex` child of a `fxLayout` parent.

I've sent in PR #466 to correct this, hopefully it can be reviewed soon. cc @ThomasBurleson

**Comment by CaerusKaru** [comment]
Closed by 8dcae02
